### PR TITLE
Add filtering functionality to bids2table

### PR DESCRIPTION
**Pull request by ReinderVosDeWael**
This PR adds filtering functionality to bids2table. It can be invoked with lists of values (that are considered an OR), or single values (equivalent to a list of a single value). See example usage below.

```python
import bids2table

bids2table.bids2table(BIDS_DIR, filters={'datatype': ['func', 'anat'], 'suffix': 'T1w'})
```

Specifically, this PR adds the following:
- A `_filter` method to `_bids2table.py`
- Unit tests for `_filter`, including a workflow to run them
- Updates the example notebook to include the new functionality
- A `exceptions.py` for custom exceptions.

### CODE REVIEW

File: bids2table/_bids2table.py [hunk 1]
```diff
@@ -112,3 +116,31 @@ def load_index(
...
+    if filters is None:
+        return df
+
+    for key, value in filters.items():
+        if not isinstance(value, list):
+            value = [value]
+        try:
+            df = df[df["entities"][key].isin(value)]
+        except KeyError as exc_info:
+            raise exceptions.InvalidFilterError(
```
Comment (clane9) [inline bids2table/_bids2table.py:142]: I don't think I would catch and re-raise a custom exception. I would just let the `KeyError` be raised. Just as informative I think and less code.

File: .github/workflows/test.yaml
Comment (clane9) [inline .github/workflows/test.yaml:1]: Let's not add CI in this PR. I'd rather do that separately.

**Review by clane9 — CHANGES_REQUESTED** [review]
Thanks for the PR @ReinderVosDeWael! Left a couple comments. And then I'll be glad to merge.

File: pyproject.toml [hunk 2]
```diff
@@ -64,6 +64,9 @@ profile = "black"
 [tool.pytest.ini_options]
 log_cli = true
 log_cli_level = "DEBUG"
+pythonpath = [
```
[the remark this answers is not in the document]
Comment (ReinderVosDeWael) [reply pyproject.toml:67]: It allows for testing without installation. It was useful locally, but I'll remove it.

File: example/example.ipynb [hunk 3]
```diff
@@ -46,10 +46,24 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "174it [00:01, 103.35it/s, tot=174, good=174, rec=2124, err=0]\n",
-      "193it [00:01, 100.70it/s, tot=193, good=193, rec=2553, err=0]\n",
-      "194it [00:01, 98.60it/s, tot=194, good=194, rec=2596, err=0]\n",
-      "219it [00:02, 101.11it/s, tot=219, good=219, rec=2914, err=0]\n"
+      "0it [00:00, ?it/s]\n",
```
[the remark this answers is not in the document]
Comment (ReinderVosDeWael) [reply example/example.ipynb]: Fixed; not sure how that sloppiness slipped through.

File: .github/workflows/test.yaml
Comment (ReinderVosDeWael) [reply .github/workflows/test.yaml:1]: Removed CI

File: example/example.ipynb [hunk 4]
```diff
@@ -39,20 +39,9 @@
...
-     "output_type": "stream",
-     "text": [
-      "174it [00:01, 103.35it/s, tot=174, good=174, rec=2124, err=0]\n",
-      "193it [00:01, 100.70it/s, tot=193, good=193, rec=2553, err=0]\n",
-      "194it [00:01, 98.60it/s, tot=194, good=194, rec=2596, err=0]\n",
-      "219it [00:02, 101.11it/s, tot=219, good=219, rec=2914, err=0]\n"
-     ]
-    }
-   ],
+   "outputs": [],
```
Comment (clane9) [inline example/example.ipynb:44]: The table generation progress log is now missing. Not sure why. Perhaps a restart and run all would fix it?

**Comment by clane9** [comment]
Superseded by #17.

**Pull request closed without merging**